ingestion/src/permissions.rs

```diff
@@ -0,0 +1,103 @@
+use crate::Error;
+use chrono::{DateTime, Utc};
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+};
+use tokio_postgres::NoTls;
+
+#[derive(Debug, Clone)]
+pub struct Permit {
+    type_id: i32,
+    param_id: i32,
+    level: i32,
+    sensor: i32,
+    from_time: DateTime<Utc>, // make sure UTC is correct for this
+    to_time: Option<DateTime<Utc>>,   // make sure UTC is correct for this
+}
+
+type StationId = i32;
+pub type PermitTable = HashMap<StationId, Vec<Permit>>;
+
+pub async fn fetch_open_permits() -> Result<PermitTable, Error> {
+    // get stinfo conn
+    let (client, conn) = tokio_postgres::connect(&std::env::var("STINFO_STRING")?, NoTls).await?;
+
+    // conn object independently performs communication with database, so needs it's own task.
+    // it will return when the client is dropped
+    tokio::spawn(async move {
+        if let Err(e) = conn.await {
+            eprintln!("connection error: {}", e); // TODO: trace this?
+        }
+    });
+
+    // query permit table
+    let rows = client
+        .query(
+            "SELECT stationid, message_formatid, paramid, hlevel, sensor, fromtime, totime \
+                FROM v_message_policy \
+                /* we're only interested in open data, which is signalled by permitid = 1 */
+                WHERE permitid = 1",
+            &[],
+        )
+        .await?;
+
+    // build hashmap
+    let mut open_permits = HashMap::new();
+
+    // indexing rows by number index instead of column name may be more performant
+    // I'm not sure if this optimisation is worth the readability tradeoff though, especially
+    // since the table if so small and this refresh is a background task
+    for row in rows {
```

Comment:
Fix error deserializing time:

thread 'main' panicked at /home/louiseo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-postgres-0.7.10/src/row.rs:151:25:
error retrieving column fromtime: error deserializing column 5: cannot convert between the Rust type `chrono::datetime::DateTime<chrono::offset::utc::Utc>` and the Postgres type `timestamp`
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

Reply:
As discussed in a video meeting, this will be addressed in a later PR for CI, not here